src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -125,7 +128,7 @@ public function preSubmit(FormEvent $event)
                 if (!$form->has($name)) {
                     $form->add($name, $this->type, array_replace(array(
                         'property_path' => '['.$name.']',
-                    ), $this->options));
+                    ), $this->getOptionsForEntry($value)));
```

Comment:
In fact I think expanded `ChoiceType` is not far from the `CollectionType` and I was thinking about decoupling the form list creation with a factory interface as #14050 did for `ChoiceType` to allow dealing with polymorphic and custom indexed/named collection.